realm/realm-jni/src/util.hpp

```diff
@@ -669,6 +671,31 @@ inline jobject NewFloat(JNIEnv* env, float value)
     return env->NewObject(java_lang_float, java_lang_float_init, value);
 }
 
+
+inline jlong to_milliseconds(realm::Timestamp ts)
+{
+    // FIXME: check for overflow
+    return 1000*ts.get_seconds() + ts.get_nanoseconds()/1000000;
+}
+
+inline realm::Timestamp from_milliseconds(jlong t)
+{
+    // FIXME: find more efficient way to convert msecs -> (secs, nsecs)
+    // FIXME: guard against overflows
+    int64_t seconds;
+    uint32_t nanoseconds;
+    if (t >= 0) {
+        seconds = int64_t(double(t)*0.001);
```

Comment:
I have changed the conversion as the `Timestamp` constructor now supports negative nanoseconds.